### PR TITLE
Calendar Import via ICS upload functional

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
This PR consists of the user being able to upload a .ics file (typically would be their Canvas Calendar feed), then the file is sent to the backend for processing (where its data is also parsed), then the parsedEvents from the file are sent back to the frontend to be displayed to the user, as well as added to the database under the logged in user.

As for bugs that encountered during development but was able to resolve, I initially was trying to approach this by trying to process and parse the .ics data in the frontend alone, but I eventually realized that wouldn't work out as intended, so I pivoted to the approach mentioned above.

Kindly find attached the test plan for this PR, showing that the user can upload an .ics file and have it appended to the list of previously existing events for that user:
https://www.loom.com/share/c60ce9884f894d4fa947611d6eb13f88?sid=e78b0ceb-d402-419d-91f3-5b352e60c047

Also something to note, I am still aware of the pending feedback from previous PRs and I intend to address them in a singular PR in order to maximize my productivity to ensure I am on track with my project timeline.

The next PR should include the implementation of the Google Calendar Sync functionality

### CODE REVIEW

File: backend/routes/calendarMgmt.js [hunk 1]
```diff
@@ -2,9 +2,16 @@ const { PrismaClient } = require('@prisma/client');
...
+const fs = require('fs');
+const ical = require('ical');
 const router = express.Router();
 const authenticateToken = require('../middlewares/authenticateToken');
 
+// Multer Middleware
+const multer = require('multer');
+const upload = multer({ dest: 'uploads/' });
+
+// Adding New Calemdar Events
```
Comment (jameskim08) [inline backend/routes/calendarMgmt.js:14]: nit: typo

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 2]
```diff
@@ -72,11 +73,19 @@ function CalendarView () {
           <p>Location: {calEvent.location}</p>
         </>
       )
-    })
+    });
+
+    const handleParsedEvents = (parsedEvents) => {
+      parsedEvents.forEach(event => {
+        createCalendarEvent(event);
```
Comment (jameskim08) [inline frontend/src/components/CalendarView/CalendarView.jsx:80]: JC suggested to move this function into a separate file. When you do this, the ICSUpload component will not need the prop in line 88, you can call it directly in it's own component.

File: frontend/src/components/ICSUpload/ICSUpload.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React, {useState} from 'react';
+
+const ICSUpload = ({onEventsImported}) => {
+    const [file, setFile] = useState(null);
+
+    const handleFileChange = (e) => {
+        setFile(e.target.files[0]);
```
Comment (jameskim08) [inline frontend/src/components/ICSUpload/ICSUpload.jsx:7]: you dont need to create a new function for simple one-site call lines like this.

File: backend/routes/calendarMgmt.js (hunk 1 above)
Comment (fiyinfoluwaafol) [reply backend/routes/calendarMgmt.js:14]: Yh, I later realized it and fixed it in a more recent PR
Thanks for pointing it out though

**Pull request merged**